### PR TITLE
[10.0] website_sale_hide_price: New module

### PR DESCRIPTION
**Pull request by chienandalu**
Website Sale Hide Price
=======================

This module allows to have hide product prices on the website store.

cc @Tecnativa

### CODE REVIEW

File: website_sale_hide_price/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Tecnativa - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    'name': 'Website Sale Hide Price',
+    'version': '10.0.1.0.0',
+    'category': 'Website',
+    'author': 'Tecnativa, '
+              'Odoo Community Association (OCA)',
+    'website': 'https://tecnativa.com',
```
Comment (yajo) [inline website_sale_hide_price/__manifest__.py]: www.

**Review by yajo — APPROVED** [review]
Code OK

File: website_sale_hide_price/README.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=======================
+Website Sale Hide Price
+=======================
+
+This module allows to have hide product prices on the website store.
```
Comment (pedrobaeza) [inline website_sale_hide_price/README.rst]: s/hide/hidden

File: website_sale_hide_price/README.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+Configuration
+=============
+
+#. Go to *Customers* and choose one.
+#. Go to *Sales and Purchases* tab.
+#. In *Sales* group set *Show prices on website* on or off so this custumer can
```
Comment (pedrobaeza) [inline website_sale_hide_price/README.rst]: s/custumer/customer

File: website_sale_hide_price/README.rst [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+   see them or not.
+
+The default value is `True`, so every partner website user can see the prices.
```
Comment (pedrobaeza) [inline website_sale_hide_price/README.rst]: Put this in the same third point

File: website_sale_hide_price/README.rst [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+Usage
+=====
+
+If no user is logged in the prices won't be shown. Prices will be shown if any
```
Comment (pedrobaeza) [inline website_sale_hide_price/README.rst]: This is also configuration information. Put a comma after `logged in`. Put here where to go to see that there are no prices.

File: website_sale_hide_price/models/res_partner.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+# Copyright 2017 Tecnativa - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    website_show_prices = fields.Boolean(
```
Comment (pedrobaeza) [inline website_sale_hide_price/models/res_partner.py]: Better to avoid plurals also in field names.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by chienandalu** [comment]
@pedrobaeza Changes done

File: website_sale_hide_price/i18n/es.po [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+"Language: es\n"
+"X-Generator: Poedit 1.8.7.1\n"
+
+#. module: website_sale_hide_price
+#: model:ir.model,name:website_sale_hide_price.model_res_partner
+msgid "Partner"
+msgstr "Empresa"
+
+#. module: website_sale_hide_price
+#: model:ir.model.fields,field_description:website_sale_hide_price.field_res_partner_website_show_prices
```
Comment (pedrobaeza) [inline website_sale_hide_price/i18n/es.po]: Translation not updated (without `s`).

File: website_sale_hide_price/i18n/es.po [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+msgid "Show prices on website"
+msgstr "Mostrar precios en el sitio web"
+
+#. module: website_sale_hide_price
+#: model:ir.model,name:website_sale_hide_price.model_website
+msgid "Website"
+msgstr "Sitio web"
+
+#. module: website_sale_hide_price
+#: model:ir.model.fields,field_description:website_sale_hide_price.field_website_website_show_prices
```
Comment (pedrobaeza) [inline website_sale_hide_price/i18n/es.po]: The same

File: website_sale_hide_price/models/website.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+from odoo.http import request
+
+
+class Website(models.Model):
+    _inherit = "website"
+
+    website_show_prices = fields.Boolean(
```
Comment (pedrobaeza) [inline website_sale_hide_price/models/website.py]: Without `s` also

File: website_sale_hide_price/views/partner_view.xml [hunk 10]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="view_partner_form" model="ir.ui.view">
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="base.view_partner_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//group[@name='sale']" position="inside">
+                <field name="website_show_prices"/>
```
Comment (pedrobaeza) [inline website_sale_hide_price/views/partner_view.xml]: Without `s`

File: website_sale_hide_price/views/website_sale_template.xml [hunk 11]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+        <template id="product_price" inherit_id="website_sale.product_price">
+            <xpath expr="//div[@itemprop='offers']" position="attributes">
+                <attribute name="t-if">
+                    website.website_show_prices
```
Comment (pedrobaeza) [inline website_sale_hide_price/views/website_sale_template.xml]: Without `s`

File: website_sale_hide_price/views/website_sale_template.xml [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+                </attribute>
+            </xpath>
+        </template>
+
+        <template id="products_item" inherit_id="website_sale.products_item">
+            <xpath expr="//div[@itemprop='offers']" position="attributes">
+                <attribute name="t-if">
+                    website.website_show_prices
```
Comment (pedrobaeza) [inline website_sale_hide_price/views/website_sale_template.xml]: Without `s`

File: website_sale_hide_price/models/website.py [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+        compute='_compute_website_show_prices')
+
+    def _compute_website_show_prices(self):
+        for rec in self:
+            if request.session.uid:
+                rec.website_show_prices =\
+                    request.env.user.partner_id.website_show_prices
```
Comment (pedrobaeza) [inline website_sale_hide_price/models/website.py]: Without `s`

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Don't do half the change...

**Comment by chienandalu** [comment]
@pedrobaeza How careless of me... :man_facepalming:

**Comment by chienandalu** [comment]
Travis seems to fail due to another module's tests

**Comment by pedrobaeza** [comment]
But can be related. Check if the tour is looking for the elements that you have modified in this one: https://github.com/OCA/e-commerce/blob/10.0/website_sale_wishlist/static/src/js/wishlist_tour.js

**Comment by chienandalu** [comment]
@pedrobaeza Indeed it hooks to a div that this module hides (https://github.com/OCA/e-commerce/blob/10.0/website_sale_wishlist/views/shop.xml#L58)

**Comment by pedrobaeza** [comment]
Then look for a way to not hide that element. I'm thinking why you put "Show prices" as option instead of "Hide prices" with default=False. Any way, the default=True should assure that the element is shown, isn't it?

**Comment by chienandalu** [comment]
@pedrobaeza I changed it to always take the partner's state so now Public user can be set on or off thus prices can be shown or hidden for anonymous users. That also should fix CIs

**Comment by chienandalu** [comment]
@pedrobaeza Everything is done here I think

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**